### PR TITLE
doc: cleanup design

### PR DESCRIPTION
**Pull request by jmolmo**
Signed-off-by: Juan Miguel Olmo Martínez <jolmomar@redhat.com>

### CODE REVIEW

File: doc/design/cleanup.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+## Main requirements
+
+Remove all the operator resource units and associated resources (config files, socket files). 
+This part is mostly complete and implemented in the ensureDeleted methods of the different resource units.
+
+In this document our target is to decide how to remove all the VGs created because operator working behaviour. 
+
+Main requirements are:
+- Volume Groups created by the operator must only be removed if we do not have PVCs or PVs using them.
+- If we decide that a VG need to be deleted, we must delete all the Logical Volumes in it (only one), and also the Physical Volume that supports the VG.
```
Comment (leelavg) [inline doc/design/cleanup.md:14]: - Operator shouldn't delete logical volumes, that'll be deleted by topolvm itself.
- Pls rephrase if your intention is something else

File: doc/design/cleanup.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+In this document our target is to decide how to remove all the VGs created because operator working behaviour. 
+
+Main requirements are:
+- Volume Groups created by the operator must only be removed if we do not have PVCs or PVs using them.
+- If we decide that a VG need to be deleted, we must delete all the Logical Volumes in it (only one), and also the Physical Volume that supports the VG. 
+- Any solution mus be compatible with the "future". (Multiples nodes, multiple lvm cluster CRDs) 
+
+## How to implement the cleanup
+
+Two possibilities has been discussed in our [first meeting](https://drive.google.com/file/d/1ltK1XO6Y3gRPVLRLvrmsBJKjXt9s2Klw/view?usp=sharing) about this topic:
```
Comment (leelavg) [inline doc/design/cleanup.md:19]: - This link will not be accessible to public, we can summarise what we discussed and mention that in this doc

File: doc/design/cleanup.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Cleanup design
+
+Cleanup: Operations needed to do when the lvm cluster CRD is removed.
+
+## Main requirements
+
+Remove all the operator resource units and associated resources (config files, socket files).
```
Comment (nbalacha) [inline doc/design/cleanup.md:7]: Does deleting the deployment/daemonset automatically remove these?

File: doc/design/cleanup.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Cleanup design
+
+Cleanup: Operations needed to do when the lvm cluster CRD is removed.
```
Comment (travisn) [inline doc/design/cleanup.md:3]: This design will be heavily influenced by discussion about the design discussion for whether we have a single LVMCluster CR, or multiple LVMVolumeGroup CRs. This current suggestion assumes the LVMCluster design. With the LVMVolumeGroup design, I see deletion steps to be the following...

When a LVMVolumeGroup is deleted:
1. There is a finalizer on the LVMVolumeGroup that prevents deletion while the cleanup is being processed.
2. The storage class for the LVMVolumeGroup is deleted, so no new PVs can be requested.
3. The operator updates the configuration for each node to signal the removal of the deleted LVMVolumeGroup
4. Each node cleans up:  
   a. If there are any LVs in the VG, block deletion
   b. If there are no LVs in the VG, delete the VG
   c. Depending on the deletion policy on the LVMVolumeGroup (which we can discuss in the future), either wipe the disk(s), or do nothing.

**Review by travisn — CHANGES_REQUESTED** [review]

**Pull request closed without merging**